docker/Dockerfile

```diff
@@ -0,0 +1,107 @@
+FROM --platform=linux/amd64 node:16.13.0 AS frontend
+
+WORKDIR /app/oppia
+
+RUN apt-get update -y && apt-get upgrade -y \
+    curl \
+    git \
+    npm
+
+# Installing packages from the package.json file
+COPY package.json .
+COPY scripts/linters/custom_eslint_checks ./scripts/linters/custom_eslint_checks
+
+RUN npm install -g yarn@1.22.15 --force
+# mathjs ESOCKETTIMEDOUT error - there are cases reported for mathjs package when timeout happens while
+# downloading the package using `yarn install` command.
+# The solution is to increase the network timeout, so that the package is downloaded successfully.
+RUN yarn install --network-timeout 1000000
+
+
+# Added platform flag here because -- https://stackoverflow.com/questions/71040681/qemu-x86-64-could-not-open-lib64-ld-linux-x86-64-so-2-no-such-file-or-direc
+FROM --platform=linux/amd64 python:3.8.17-slim-buster AS backend
+
+WORKDIR /
+
+ENV OPPIA_IS_DOCKERIZED="true"
+
+# Installing the pre-requisites libs and dependencies
+RUN apt-get update -y && apt-get upgrade -y \
+    curl \
+    npm \
+    openjdk-11-jre \
+    python2 \
+    python3-matplotlib \
+    unzip \
+    wget
+
+RUN pip install --upgrade pip==21.2.3
+RUN pip install pip-tools==6.6.2 setuptools==58.5.3 cmake
+
+# Installing google cloud sdk
+RUN wget -O gcloud-sdk.tar.gz https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-364.0.0-linux-x86_64.tar.gz && \
+    tar -xzf gcloud-sdk.tar.gz && \
+    ./google-cloud-sdk/install.sh --quiet && \
+    rm gcloud-sdk.tar.gz
+
+WORKDIR /app/oppia
+
+# Installing python dependencies from the requirements_dev.txt file
+COPY requirements_dev.txt .
+
+RUN pip install -r requirements_dev.txt
+
+# Installing buf and proto for Linux -- this docker container is based on Linux
+COPY buf.gen.yaml .
+COPY /extensions/classifiers/proto/ ./extensions/classifiers/proto/
+
+# Installing third party dependencies
+COPY scripts ./scripts
+COPY /core/utils.py ./core/utils.py
+COPY /core/feconf.py ./core/feconf.py
+COPY /core/constants.py ./core/constants.py
+COPY /assets/constants.ts ./assets/constants.ts
+COPY /assets/release_constants.json ./assets/release_constants.json
```

Comment:
this will create so many layers in the images, do we still need this considering we are running build.py from entrypoint? Can we simplify it?

Reply:
these files are not required for build.py they are for the install_dependencies script

Reply:
done